### PR TITLE
Implement full vault backup and restore

### LINKED ISSUE #11975: Implement full vault backup and restore
Implement full vault backup and restore based on https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2/backuprestore.json. Note that restore should probably be implemented as an LRO, which would poll the "pending" endpoint instead of exposing it outright. I have Azure/azure-rest-api-specs#9806 opened to track.

### LINKED ISSUE #13457: Ship Key Vault Administration package as 4.0.0-beta.1
Based on discussion with architects, we'll want to ship the administration package as a 4.0 since it's branch new. It will not be in version lock-step with other Key Vault packages.

### PR DESCRIPTION
**Pull request by chlowell**
Closes #11975 by adding `KeyVaultBackupClient`, which performs full vault backup and restore operations, and selective key restore operations. Also sets the package version to 4.0.0b1 (closes #13457).

I'm opening this as a draft because I'm still working on tests. I have (passing!) tests locally but they rely on ugly workarounds for the difficulties of testing managed HSM and their recordings aren't yet safe to commit.

### CODE REVIEW

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            azure_storage_blob_container_uri=sas_parameter,
+            cls=BackupOperation._wrap_generated,
+            continuation_token=kwargs.pop("continuation_token", None),
+            polling=LROBasePolling(lro_algorithms=[KeyVaultBackupClientPolling()], timeout=polling_interval, **kwargs),
+            **kwargs
+        )
+
+    def begin_full_restore(self, blob_storage_uri, sas_token, folder_name, **kwargs):
+        # type: (str, str, str, **Any) -> LROPoller[RestoreOperation]
+        """Restore a Key Vault backup.
```
Comment (iscai-msft) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py:51]: tiniest nit #
```suggestion
        """Restore a full backup of a Key Vault
```

**Review by iscai-msft — APPROVED** [review]

**Review by christothes — APPROVED** [review]
Looks good at the api level

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py [hunk 2]
```diff
@@ -212,4 +212,14 @@ class RestoreOperation(_Operation):
...
 
 
-SelectiveKeyRestoreOperation = RestoreOperation
+class SelectiveKeyRestoreOperation(_Operation):
+    """A Key Vault operation restoring a single key.
+
+    :ivar str status: status of the backup operation
+    :ivar str status_details: more details of the operation's status
+    :ivar error: Error encountered, if any, during the operation
+    :type error: ~key_vault_client.models.Error
```
Comment (iscai-msft) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py:221]: don't think this typing would render properly

**Review by iscai-msft — DISMISSED** [review]

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py (hunk 2 above)
Comment (chlowell) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py:221]: Right you are. I copied it from the generated model. Fixing it depends on how we expose the error. I made this part of #13575.

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_backup_client.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            polling=AsyncLROBasePolling(
+                lro_algorithms=[KeyVaultBackupClientPolling()], timeout=polling_interval, **kwargs
+            ),
+            **kwargs
+        )
+
+    async def begin_full_restore(
+        self, blob_storage_uri: str, sas_token: str, folder_name: str, **kwargs: "Any"
+    ) -> "AsyncLROPoller[RestoreOperation]":
+        """Restore a Key Vault backup.
```
Comment (mccoyp) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_backup_client.py:55]: nit per the comment on [the sync version](https://github.com/Azure/azure-sdk-for-python/pull/13525/files#r482486160)
```suggestion
        """Restore a full backup of a Key Vault.
```

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py [hunk 4]
```diff
@@ -165,3 +165,61 @@ def _from_generated(cls, definition):
...
+
+    def __init__(self, **kwargs):
+        self.azure_storage_blob_container_uri = kwargs.pop("azure_storage_blob_container_uri", None)
+        super(BackupOperation, self).__init__(**kwargs)
+
+
+class RestoreOperation(_Operation):
+    """A Key Vault restore operation.
+
+    :ivar str status: status of the backup operation
```
Comment (mccoyp) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py:205]: Should this be the status of the restore operation?

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py [hunk 5]
```diff
@@ -165,3 +165,61 @@ def _from_generated(cls, definition):
...
+    :ivar datetime.datetime start_time: UTC start time of the operation
+    :ivar datetime.datetime end_time: UTC end time of the operation
+    :ivar str job_id: identifier for the operation
+    """
+
+
+class SelectiveKeyRestoreOperation(_Operation):
+    """A Key Vault operation restoring a single key.
+
+    :ivar str status: status of the backup operation
```
Comment (mccoyp) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py:218]: Same question about restore vs. backup operation

**Review by mccoyp — DISMISSED** [review]

**Review by iscai-msft — APPROVED** [review]

**Pull request merged**